pages/languages.js

```diff
@@ -1,69 +1,90 @@
-import styled from "styled-components";
+import React, { useEffect, useState } from "react";
 import Link from "next/link";
+import { useRouter } from "next/router";
 import {
+  Box,
   ContainerMain,
   ContainerMainLabel,
-  Box,
   InnereBox,
   SpaceLabel,
-} from "@/components/styles/container.styled";
-import { useState } from "react";
+} from "@/components/styledComponents/Container.styled";
 
 export default function PageLanguage() {
-  // the language is setted as default direct here
-  const [languages, setLanguages] = useState("english");
+  const [language, setLanguage] = useState("");
 
-  //function finalLanguage() {}
+  useEffect(() => {
+    const storedLanguage = localStorage.getItem("language");
+    if (storedLanguage) {
+      setLanguage(storedLanguage);
+    } else {
+      setLanguage("english");
+      localStorage.setItem("language", "english");
+    }
+  }, []);
 
-  function handleCheckLanguage(e) {
-    setLanguages(e);
+  function handleLanguage(event) {
+    const { value } = event.target;
+    setLanguage(value);
+    localStorage.setItem("language", value);
+    const storedLanguage = localStorage.getItem("language");
+    console.log("the lang is " + storedLanguage);
   }
 
-  console.log(languages);
+  useEffect(() => {
+    if (language === "") {
+      setLanguage("english");
+    }
+  }, [language]);
```

Comment:
Please do not roll your own local storage state implementation. Connecting state to local storage correctly is very hard and can lead to strange and hard to find bugs.

Please use the `use-local-storage-state` package we also used in the sessions. Check the the handouts for React with Local Storage to find out how to use it.

Basically all the code here can be simplified to:

```js
const [language, setLanguage] = useLocalStorageState("language", { defaultValue: "english" });
```